client/extensions/woocommerce/app/promotions/promotion-header.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+import { isObject } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ActionHeader from 'woocommerce/components/action-header';
+import Button from 'components/button';
+import { getLink } from 'woocommerce/lib/nav-utils';
+
+function renderTrashButton( onTrash, promotion, isBusy, translate ) {
+	return onTrash && (
+		<Button borderless scary onClick={ onTrash }>
+			<Gridicon icon="trash" />
+			<span>{ translate( 'Delete' ) }</span>
+		</Button>
+	);
+}
+
+function renderSaveButton( onSave, promotion, isBusy, translate ) {
+	const saveExists = 'undefined' !== typeof onSave;
+	const saveDisabled = false === onSave;
+
+	const saveLabel = ( promotion && ! isObject( promotion.id )
+		? translate( 'Update' )
+		: translate( 'Save & Publish' )
+	);
+
+	return saveExists && (
+		<Button primary onClick={ onSave } disabled={ saveDisabled } busy={ isBusy }>
+			{ saveLabel }
+		</Button>
+	);
+}
+
+const PromotionHeader = ( { promotion, onSave, onTrash, isBusy, translate, site } ) => {
+	const existing = promotion && ! isObject( promotion.id );
+
+	const trashButton = renderTrashButton( onTrash, promotion, isBusy, translate );
+	const saveButton = renderSaveButton( onSave, promotion, isBusy, translate );
+
+	const currentCrumb = promotion && existing
```

Comment:
Minor: Since we don't need `existing` for anything else we could just do the `! isObject` check here and save a line.

Reply:
Fair enough, changed.